src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -54,7 +54,15 @@ protected override void AttachCore()
             if ((this.Mode & PeerAddressManagerBehaviourMode.Discover) != 0)
             {
                 if (this.AttachedPeer.State == NetworkPeerState.Connected)
+                {
+                    //If a peer connects via the peer discovery loop but does not yet exist
```

Comment:
hmm I didnt understand this can you explain more?
When we find addresses using address discovery we add them to the address manger right?
Then the discover peer loop will try to connect so this should already be in the address manager.

Reply:
@dangershony OK there are 2 scenarios, peer discovery with no peers file and peer discovery with peers file. The first scenario we have to get addresses from the seed nodes, which when one them is connected to, they aren't yet in the peer address manager's list. This is why we need to add them here if they don't exist.

I had a look at the old AddressManager:

[Old Discover Loop](https://github.com/stratisproject/StratisBitcoinFullNode/commit/50f4cd76aff090a4ddcf056803485d745b31d684#diff-5eabda437d068ea8bbe924ebee5950d9R1130)

And it behaved the same way 👍